### PR TITLE
Fix tests in src/test/isolation2/expected/distributed_snapshot.out

### PR DESCRIPTION
**Pull request by RekGRpth**
Commit 623a9ba in src/backend/storage/ipc/procarray.c in the
GetSnapshotData function enables snapshot reuse. This omits the call to
the DistributedLog_AdvanceOldestXmin function below, which sets the
distributedlog_advance_oldest_xmin fault injector used in the
src/test/isolation2/sql/distributed_snapshot.sql test. Skip snapshot
data reuse for this test.

### CODE REVIEW

**Comment by silent-observer** [comment]
How valid is the test itself when reusing snapshots? It seems to me like the test tries to create a specific edge case for distributed snapshots, so if the oldestXmin is not actually advanced when the test expects, it might not be able to reproduce the original problem anymore. Maybe we should instead avoid snapshot reuse for this particular test?

**Comment by KnightMurloc** [comment]
This test checks the work with `globalXmin`, which no longer exists. In general, I don't really understand what the first scenario is testing and what its fail should look like.

**Review by bandetto — APPROVED** [review]

**Review by silent-observer — APPROVED** [review]

**Comment by RekGRpth** [comment]
reworked

File: src/backend/storage/ipc/procarray.c [hunk 1]
```diff
@@ -2812,6 +2812,18 @@ GetSnapshotData(Snapshot snapshot, DtxContext distributedTransactionContext)
 	 */
 	LWLockAcquire(ProcArrayLock, LW_SHARED);
 
+#ifdef FAULT_INJECTOR
+	if (!IS_QUERY_DISPATCHER() && snapshot->haveDistribSnapshot &&
+		FaultInjector_InjectFaultIfSet("distributedlog_advance_oldest_xmin_check",
```
Comment (silent-observer) [inline src/backend/storage/ipc/procarray.c]: Maybe name it something more suggestive, like "distributed_snapshot_skip_reuse"? It could be that we would need this fault injector somewhere else in the future

Comment (RekGRpth) [reply src/backend/storage/ipc/procarray.c]: renamed

**Review by KnightMurloc — APPROVED** [review]

**Review by silent-observer — APPROVED** [review]
